src/coreclr/utilcode/clrhost_nodependencies.cpp

```diff
@@ -318,39 +253,10 @@ operator new(size_t n)
 void * __cdecl
 operator new[](size_t n)
 {
-#ifdef _DEBUG_IMPL
-    CLRThrowsExceptionWorker();
-#endif
-
-    STATIC_CONTRACT_THROWS;
-    STATIC_CONTRACT_GC_NOTRIGGER;
-    STATIC_CONTRACT_FAULT;
-    STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
-
-    void* result = ClrMalloc(n);
-    if (result == NULL) {
-        ThrowOutOfMemory();
-    }
-    TRASH_LASTERROR;
-    return result;
-};
-
-
-void * __cdecl operator new(size_t n, const std::nothrow_t&) noexcept
-{
-    STATIC_CONTRACT_NOTHROW;
-    STATIC_CONTRACT_GC_NOTRIGGER;
-    STATIC_CONTRACT_FAULT;
-    STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
-
-    INCONTRACT(_ASSERTE(!ARE_FAULTS_FORBIDDEN()));
-
-    void* result = ClrMalloc(n);
-	TRASH_LASTERROR;
-    return result;
+    return ::operator new(n);
```

Comment:
I am wondering why this was manually inlined before. Is this wrapper going to introduce an extra path length?

Reply:
I would expect this to be inlined or tail-called with our optimization settings.